mapie/quantile_regression.py

```diff
@@ -0,0 +1,425 @@
+        ValueError
+            If the value of alpha is not between 0 and 0.5.
+        """
+        if isinstance(alpha, float):
+            if np.any(np.logical_or(alpha <= 0, alpha >= 0.5)):
+                raise ValueError(
+                    "Invalid alpha. Allowed values are between 0 and 0.5."
+                )
+            else:
+                alpha_np = np.array([alpha / 2, 1 - alpha / 2, 0.5])
+        else:
+            raise ValueError(
+                "Invalid alpha. Allowed values are float."
+            )
+        return alpha_np
+
+    def _check_estimator(
+        self,
+        estimator: Optional[Union[RegressorMixin, Pipeline]] = None
+    ) -> Union[RegressorMixin, Pipeline]:
+        """
+        Perform several checks on the estimator to check if it has
+        all the required specifications to be used with this methodology.
+
+        Parameters
+        ----------
+        estimator : Optional[RegressorMixin], optional
+            Estimator to check, by default ``None``.
+
+        Returns
+        -------
+        RegressorMixin
+            The estimator itself or a default ``QuantileRegressor`` instance
+            with ``solver`` set to "highs".
+
+        Raises
+        ------
+        ValueError
+            If the estimator fit or predict methods.
+        ValueError
+            We check if it's a known estimator that does quantile regression
+            according to the dictionnary set quantile_estimator_params.
+            This dictionnary will need to be updated with the latest new
+            available estimators.
+        ValueError
+            The estimator does not have the "loss_name" in its parameters and
+            therefore can not be used as an estimator.
+        ValueError
+            There is no quantile "loss_name" and therefore this estimator
+            can not be used as a ``MapieQuantileRegressor``.
+        ValueError
+            The parameter to set the alpha value does not exist in this
+            estimator and therefore we cannot use it.
+        """
+        if estimator is None:
+            return QuantileRegressor(
+                solver="highs-ds",
+                alpha=1e-9,
+            )
+        if not (hasattr(estimator, "fit") and hasattr(estimator, "predict")):
+            raise ValueError(
+                "Invalid estimator. "
+                "Please provide a regressor with fit and predict methods."
+            )
+        if isinstance(estimator, Pipeline):
+            self._check_estimator(estimator[-1])
+            return estimator
+        else:
+            name_estimator = estimator.__class__.__name__
+            if name_estimator == "QuantileRegressor":
+                return estimator
+            else:
+                if name_estimator in self.quantile_estimator_params:
+                    param_estimator = estimator.get_params()
+                    loss_name, alpha_name = self.quantile_estimator_params[
```

Comment:
Correct syntax :
```
self.quantile_estimator_params[
    name_estimator
].values()
```

Reply:
done

Reply:
Are you sure about the indentation of the last line ?

Reply:
yes, done in last commit